### PR TITLE
Add libconfini

### PR DESCRIPTION
**Pull request by madmurphy**
Hi Sean!

I am the author of [libconfini](https://github.com/madmurphy/libconfini), a smart, independent, two-file INI parser able to read several formats of INI files. Would you like to add it to your list?

--madmurphy

### CODE REVIEW

**Comment by nothings** [comment]
It's 3 files because it requires COPYING as well.

**Comment by madmurphy** [comment]
Hi Sean! Thanks for your answer.

But what about the libraries oosmos, mongoose and SLRE? Don't they use the GPL license as well? Technically the library does not need a copy of the GPL license, it's _who distributes it_ who does.

**Comment by madmurphy** [comment]
Hi Sean! Thanks for your answer.

But what about the libraries oosmos, mongoose and SLRE? Don't they use the GPL license as well? Technically the library does not need a copy of the GPL license, it's _who distributes it_ who does. But the final user is not forced to download the license as well.

--madmurphy

**Comment by nothings** [comment]
Both mongoose and SLRE do not require a separate COPYING be redistributed; both have a LICENSE file which is redundant as the text of LICENSE is included in one of the source files as well, which means there is no need to separately keep a LICENSE/COPYING file as well. oosmos was included by error.

Nobody knows when they start a project whether they might want to make the source available at some point. If you download a third-party library and put it into your source without accounting for the redistribution limitations, you're just asking for a world of hurt in the future. Thus the technicality doesn't interest me.

**Comment by madmurphy** [comment]
If you use it as a shared library, you should not worry about any license, since you rely on the fact that the user has it already installed on his computer (with a copy of the license or not – as I said before the final user does not need the license unless he wants it).

If you embed it in your code instead, since this library is a GPL library, it can be included only within a GPL application, which means the application itself must have its own GPL LICENSE file, and that will be sufficient for both your application and the embedded library.

So, in short, I find a bit weak to claim that this is not a two-file library. But thank you anyway for your consideration.

--madmurphy

**Comment by nothings** [comment]
Please read the section 'New libraries and corrections' immediately at the end of the list. This rule has been in place since the list was first created, and I don't give a shit what you 'find a bit weak'.
